client/consensus/slots/src/lib.rs

```diff
@@ -242,6 +250,51 @@ pub trait SimpleSlotWorker<B: BlockT> {
 		Some(proposal)
 	}
 
+	/// Calls `create_inherent_data` and handles errors.
+	async fn create_inherent_data(
+		slot_info: &SlotInfo<B>,
+		logging_target: &str,
+	) -> Option<sp_inherents::InherentData> {
+		let now = Instant::now();
+		let remaining_duration = if now > slot_info.ends_at {
```

Comment:
This is basically saturating_sub

Reply:
Right, as what sub for `Instant` does for now but might not in the future https://doc.rust-lang.org/stable/std/time/struct.Instant.html#method.sub-1

Reply:
I think Basti meant to use https://doc.rust-lang.org/std/time/struct.Instant.html#method.saturating_duration_since instead of if else block

Reply:
Thanks, much clearer